### PR TITLE
refactor(sbb-navigation): lit migration

### PR DESCRIPTION
**Pull request by dauriamarco**
Migrate `sbb-navigation`, `sbb-navigation-section`, `sbb-navigation-marker`, `sbb-navigation-list` and `sbb-navigation-action` to Lit.

### CODE REVIEW

File: src/components/sbb-navigation-marker/sbb-navigation-marker.stories.tsx [hunk 1]
```diff
@@ -16,7 +18,7 @@ const defaultArgTypes: ArgTypes = {
 };
 
 const defaultArgs: Args = {
-  size: size.options[0],
+  // size: size.options[0],
```
Comment (TomMenga) [inline src/components/sbb-navigation-marker/sbb-navigation-marker.stories.tsx:21]: Is this meant? or just a mistake?

File: src/components/sbb-navigation-marker/sbb-navigation-marker.tsx [hunk 2]
```diff
@@ -55,73 +49,81 @@ export class SbbNavigationMarker implements ComponentInterface {
...
-  @Method()
-  public async select(action: HTMLSbbNavigationActionElement): Promise<void> {
-    await this.reset();
+  public select(action: SbbNavigationAction): void {
+    this.reset();
     action.active = true;
     this._currentActiveAction = action;
     this._hasActiveAction = true;
-    this._setMarkerPosition();
+    setTimeout(() => this._setMarkerPosition());
```
Comment (TomMenga) [inline src/components/sbb-navigation-marker/sbb-navigation-marker.tsx:68]: 😢

File: src/components/sbb-navigation-section/sbb-navigation-section.tsx [hunk 3]
```diff
@@ -326,79 +323,89 @@ export class SbbNavigationSection implements ComponentInterface {
...
-    const backButton = (
+  protected override render(): TemplateResult {
+    const backButton = html`
       <sbb-button
         class="sbb-navigation-section__back"
-        arial-label={this.accessibilityBackLabel || i18nGoBack[this._currentLanguage]}
+        arial-label=${this.accessibilityBackLabel || i18nGoBack[this._currentLanguage]}
         variant="transparent"
-        negative={true}
+        negative=${true}
```
Comment (TomMenga) [inline src/components/sbb-navigation-section/sbb-navigation-section.tsx:347]: Isn't this just `negative`?

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 4]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
 
   it('renders', () => {
-    expect(element).toHaveClass('hydrated');
+    assert.instanceOf(element, SbbNavigation);
   });
 
   it('opens the navigation', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const dialog = await page.find('sbb-navigation >>> dialog');
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:43]:
```suggestion
    const didOpenEventSpy = new EventSpy(events.didOpen);
```

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 5]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
+    expect(element).to.have.attribute('data-state', 'opened');
+    expect(dialog).to.have.attribute('open');
   });
 
   it('closes the navigation', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const didCloseEventSpy = await page.spyOnEvent(events.didClose);
-    const dialog = await page.find('sbb-navigation >>> dialog');
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
+    const didCloseEventSpy = new EventSpy(events.didClose, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:59]:
```suggestion
    const didOpenEventSpy = new EventSpy(events.didOpen);
    const didCloseEventSpy = new EventSpy(events.didClose);
```

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 6]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
+    expect(dialog).not.to.have.attribute('open');
   });
 
   it('closes the navigation on close button click', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const didCloseEventSpy = await page.spyOnEvent(events.didClose);
-    const dialog = await page.find('sbb-navigation >>> dialog');
-    const closeButton = await page.find('sbb-navigation >>> .sbb-navigation__close');
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
+    const didCloseEventSpy = new EventSpy(events.didClose, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:83]:
```suggestion
    const didOpenEventSpy = new EventSpy(events.didOpen);
    const didCloseEventSpy = new EventSpy(events.didClose);
```

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 7]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
-    expect(dialog).not.toHaveAttribute('open');
+    expect(dialog).not.to.have.attribute('open');
   });
 
   it('closes the navigation on Esc key press', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const didCloseEventSpy = await page.spyOnEvent(events.didClose);
-    const dialog = await page.find('sbb-navigation >>> dialog');
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
+    const didCloseEventSpy = new EventSpy(events.didClose, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:108]: Same thing

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 8]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
-      'sbb-navigation > sbb-navigation-marker > sbb-navigation-action[sbb-navigation-close]',
-    );
-
-    await element.callMethod('open');
-    await page.waitForChanges();
-
-    action.triggerEvent('click');
-    await page.waitForChanges();
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
+    const didCloseEventSpy = new EventSpy(events.didClose, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:135]: Same thing

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 9]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
+    expect(dialog).not.to.have.attribute('open');
+    expect(sectionDialog).not.to.have.attribute('open');
   });
 
   it('opens navigation and opens section', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const navDialog = await page.find('sbb-navigation >>> dialog');
-    const sectionDialog = await page.find('sbb-navigation-section#first-section >>> dialog');
-    const action = await page.find(
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:172]: same

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 10]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const navDialog = await page.find('sbb-navigation >>> dialog');
-    const firstSectionDialog = await page.find('sbb-navigation-section#first-section >>> dialog');
-    const secondSectionDialog = await page.find('sbb-navigation-section#second-section >>> dialog');
-    const firstAction = await page.find('sbb-navigation-marker > sbb-navigation-action#action-1');
-    const secondAction = await page.find('sbb-navigation-marker > sbb-navigation-action#action-2');
-
-    await element.callMethod('open');
-    await page.waitForChanges();
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:199]: sam

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 11]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
+    expect(secondSectionDialog).not.to.have.attribute('open');
 
-    secondAction.triggerEvent('click');
-    await page.waitForChanges();
+    secondAction.click();
+    await element.updateComplete;
 
-    expect(firstSectionDialog).not.toHaveAttribute('open');
-    expect(secondSectionDialog).toHaveAttribute('open');
+    expect(firstSection.getAttribute('data-state')).not.to.be.equal('opened');
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:233]:
```suggestion
    expect(firstSection).not.to.have.attribute('data-state', 'opened');
```

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 12]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
   });
 
   it('closes the navigation and the section on close button click', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const didCloseEventSpy = await page.spyOnEvent(events.didClose);
-    const dialog = await page.find('sbb-navigation >>> dialog');
-    const sectionDialog = await page.find('sbb-navigation-section#first-section >>> dialog');
-    const action = await page.find(
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
+    const didCloseEventSpy = new EventSpy(events.didClose, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:239]: same

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 13]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
   });
 
   it('closes the navigation and the section on Esc key press', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const didCloseEventSpy = await page.spyOnEvent(events.didClose);
-    const dialog = await page.find('sbb-navigation >>> dialog');
-    const sectionDialog = await page.find('sbb-navigation-section#first-section >>> dialog');
-    const action = await page.find(
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
+    const didCloseEventSpy = new EventSpy(events.didClose, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:275]: sam

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 14]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
+    expect(dialog).not.to.have.attribute('open');
+    expect(sectionDialog).not.to.have.attribute('open');
   });
 
   it('closes section with sbb-navigation-section-close', async () => {
-    const didOpenEventSpy = await page.spyOnEvent(events.didOpen);
-    const dialog = await page.find('sbb-navigation >>> dialog');
-    const sectionDialog = await page.find('sbb-navigation-section#first-section >>> dialog');
-    const action = await page.find(
+    const didOpenEventSpy = new EventSpy(events.didOpen, document);
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:312]: image

File: src/components/sbb-navigation/sbb-navigation.e2e.ts [hunk 15]
```diff
@@ -26,295 +33,310 @@ describe('sbb-navigation', () => {
...
 
-    await closeEl.click();
-    await page.waitForChanges();
+    closeEl.click();
+    await element.updateComplete;
 
-    expect(dialog).toHaveAttribute('open');
-    expect(sectionDialog).not.toHaveAttribute('open');
+    expect(dialog).to.have.attribute('open');
+    expect(section.getAttribute('data-state')).not.to.be.equal('opened');
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.e2e.ts:340]:
```suggestion
    expect(section).not.to.have.attribute('data-state', 'opened');
```

File: src/components/sbb-navigation/sbb-navigation.tsx [hunk 16]
```diff
@@ -52,87 +44,102 @@ let nextId = 0;
...
-  @State() private _currentLanguage = documentLanguage();
+  @state() private _currentLanguage = documentLanguage();
 
   /**
    * Emits whenever the navigation begins the opening transition.
    */
-  @Event({
+  private _willOpen: EventEmitter<void> = new EventEmitter(this, events.willOpen, {
     bubbles: true,
     composed: true,
```
Comment (TomMenga) [inline src/components/sbb-navigation/sbb-navigation.tsx:109]:
```suggestion
```

**Review by TomMenga — COMMENTED** [review]
Good job 😄 , some minor improvements

File: src/components/sbb-navigation/sbb-navigation.e2e.ts (hunk 14 above)
Comment (dauriamarco) [reply src/components/sbb-navigation/sbb-navigation.e2e.ts:312]: image

File: src/components/sbb-navigation-action/sbb-navigation-action.tsx [hunk 17]
```diff
@@ -1,159 +1,167 @@
...
    * The type attribute to use for the button.
    */
-  @Prop() public type: ButtonType | undefined;
+  @property() public type: ButtonType | undefined;
 
   /**
    * Whether the action is active.
    */
-  @Prop({ reflect: true }) public active = false;
+  @property({ reflect: true, type: Boolean })
```
[the remark this answers is not in the document]
Comment (dauriamarco) [reply src/components/sbb-navigation-action/sbb-navigation-action.tsx:66]: Yes, it should be.

File: src/components/sbb-navigation-marker/sbb-navigation-marker.tsx (hunk 2 above)
Comment (dauriamarco) [reply src/components/sbb-navigation-marker/sbb-navigation-marker.tsx:68]: I know, unfortunately here if I do not use the timeout the query `this.shadowRoot.querySelector('[data-active]')` fails and does not work...

**Review by TomMenga — APPROVED** [review]
LGTM 👍

**Pull request merged**